File: examples/secure_sum_r/bob.r
library(devtools)
source_url("https://raw.githubusercontent.com/fdrtd/client-r/main/src/fdrtd.r")

source("shared.r")

SECRET_BOB <- 16.0
NETWORK_BOB <- list(nodes=NODES, myself=1)    # Bob is no. 1 out of 0, 1, 2.

api <- Api(URL_BOB)
microservice <- api$create(kwargs=list(protocol="Simon"))
compute <- microservice$attribute("compute")
result <- compute$call(list(), list(microprotocol="SecureSum", data=SECRET_BOB, network=NETWORK_BOB, uuid=UUID))
print(api$download(result))

File: examples/secure_sum_python/bob.py
"""
secure sum example with SIMON (SImple Multiparty computatiON)

1. run fdrtd servers (see shared.py)
2. run alice.py, bob.py, and charlie.py simultaneously
"""


import representation
import shared


SECRET_BOB = 16.0  # Bob's secret input
NETWORK_BOB = {**shared.NETWORK, 'myself': 1}  # Bob's server is node #1 of the shared network


if __name__ == "__main__":

    api = representation.Api(shared.URL_BOB)
    microservice = api.create(protocol="Simon")
    result = microservice.compute(microprotocol="SecureSum",
                                  data=SECRET_BOB,
                                  network=NETWORK_BOB,
                                  uuid=shared.UUID)
    print(api.download(result))
